### PR TITLE
K8SPG-373: Create default user manually

### PR DESCRIPTION
**Pull request by inelpandzic**
[K8SPG-373](https://jira.percona.com/browse/K8SPG-373) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
  We rely on Crunchy to create the default `<cluster-name>-pguser-<cluster-name>` user, but it will not be created if we create a cluster with PMM enabled, because if `spec.users` is populated during cluster creation, Crunchy does not create this default user.

**Solution:**
Create `<cluster-name>-pguser-<cluster-name>` manually and don't rely on Crunchy to create it if PMM is enabled and there are no users specified in `spec.users`.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [x] Is an E2E test/test case added for the new feature/change?
- [x] Are unit tests added where appropriate?

**Config/Logging/Testability**
- [x] Are all needed new/changed options added to default YAML files?
- [x] Are the manifests (crd/bundle) regenerated if needed?
- [x] Did we add proper logging messages for operator actions?
- [x] Did we ensure compatibility with the previous version or cluster upgrade process?
- [x] Does the change support oldest and newest supported PG version?
- [x] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/753df54a8/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/753df54a85ae9ff3fc3501b1a3c9b376eb27d8a6
image: `perconalab/percona-postgresql-operator:PR-464-753df54a8`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/d6c34c756/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/d6c34c756f674dd23376a949271f02aacd004670
image: `perconalab/percona-postgresql-operator:PR-464-d6c34c756`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/5774c28472bdbfc01d92f5f8df460c601a88e6c4
image: `perconalab/percona-postgresql-operator:PR-464-5774c2847`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/5774c28472bdbfc01d92f5f8df460c601a88e6c4
image: `perconalab/percona-postgresql-operator:PR-464-5774c2847`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/5774c2847/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/5774c28472bdbfc01d92f5f8df460c601a88e6c4
image: `perconalab/percona-postgresql-operator:PR-464-5774c2847`

**Comment by nmarukovich** [comment]
It looks like we can create cluster with name monitor and it leads to the problem when we enable pmm after cluster creation. 
Could we prohibit to create cluster with name which is equal some predefined operator users? @inelpandzic

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/451d27da4/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/451d27da4fc184532e0297475105882121e0ec9b
image: `perconalab/percona-postgresql-operator:PR-464-451d27da4`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/demand-backup.log) |
| scaling | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/start-from-backup.log) |
| monitoring | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/monitoring.log) |
| telemetry-transfer | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/40fb0e233/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/40fb0e2337be60b98171b7581e7892216604492a
image: `perconalab/percona-postgresql-operator:PR-464-40fb0e233`

**Comment by inelpandzic** [comment]
> It looks like we can create cluster with name monitor and it leads to the problem when we enable pmm after cluster creation. Could we prohibit to create cluster with name which is equal some predefined operator users? @inelpandzic If I created monitor cluster and then made pmm true, I got an error: `time="2023-06-05T10:36:23Z" level=error msg="Reconciler error" PerconaPGCluster=pgo-user/monitor controller=perconapgcluster controllerGroup=[pgv2.percona.com](http://pgv2.percona.com/) controllerKind=PerconaPGCluster error="update/create PostgresCluster: [PostgresCluster.postgres-operator.crunchydata.com](http://postgrescluster.postgres-operator.crunchydata.com/) \"monitor\" is invalid: spec.users[1]: Duplicate value: map[string]interface {}{\"name\":\"monitor\"}" file="percona/controller/pgcluster/controller.go:234" func="pgcluster.(*PGClusterReconciler).Reconcile" name=monitor namespace=pgo-user reconcileID=ab22d2b8-883f-4838-9b5e-6db37a47dc41 version=`

[d4201e1](https://github.com/percona/percona-postgresql-operator/pull/464/commits/d4201e17d11ce5f69eefdc2abfeed4fd51928f44)

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/7e2be6961/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/7e2be6961f3aef09a4387317615445ec15ce54db
image: `perconalab/percona-postgresql-operator:PR-464-7e2be6961`

**Review by tplavcic — DISMISSED** [review]

File: percona/controller/pgcluster/controller.go [hunk 1]
```diff
@@ -205,6 +205,19 @@ func (r *PGClusterReconciler) Reconcile(ctx context.Context, request reconcile.R
...
 				},
 			})
+
+			if cr.Spec.Users == nil || len(cr.Spec.Users) == 0 {
+				// Add default user: <cluster-name>-pguser-<cluster-name>
+				postgresCluster.Spec.Users = append(postgresCluster.Spec.Users, v1beta1.PostgresUserSpec{
+					Name: v1beta1.PostgresIdentifier(cr.Name),
+					Databases: []v1beta1.PostgresIdentifier{
+						v1beta1.PostgresIdentifier(cr.Name),
+					},
```
Comment (egegunes) [inline percona/controller/pgcluster/controller.go:215]: is this the default behavior?

**Review by egegunes — APPROVED** [review]

File: percona/controller/pgcluster/controller.go (hunk 1 above)
Comment (inelpandzic) [reply percona/controller/pgcluster/controller.go:215]: Yes

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/520bc1928/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/520bc19283598abbd2bcf75f8e9a0b4f91148ff7
image: `perconalab/percona-postgresql-operator:PR-464-520bc1928`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/telemetry-transfer.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/87c35d665d83e8eb6c0180b6893106dd2dd95bbd
image: `perconalab/percona-postgresql-operator:PR-464-87c35d665`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/init-deploy.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/demand-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/scaling.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/scheduled-backup.log) |
| start-from-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/start-from-backup.log) |
| monitoring | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/monitoring.log) |
| telemetry-transfer | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/telemetry-transfer.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pg-operator/PR-464/87c35d665/users.log) |
| We run 8 out of 8|

commit: https://github.com/percona/percona-postgresql-operator/pull/464/commits/87c35d665d83e8eb6c0180b6893106dd2dd95bbd
image: `perconalab/percona-postgresql-operator:PR-464-87c35d665`

**Review by tplavcic — APPROVED** [review]

**Review by egegunes — APPROVED** [review]

**Pull request merged**